src/types.ts

```diff
@@ -7,13 +7,75 @@ import {
   StructDefinition,
   VariableDeclaration,
 } from 'solc-typed-ast';
+import { Static, Type } from '@sinclair/typebox';
 
-export interface Config {
-  include: string; // Required: Glob pattern of files to process.
-  exclude: string; // Optional: Glob pattern of files to exclude.
-  root: string; // Optional: Project root directory.
-  enforceInheritdoc: boolean; // Optional: True if all external and public functions should have @inheritdoc.
-  constructorNatspec: boolean; // Optional: True if the constructor should have natspec.
+// NOTE: For params like `return` if its set to true we will only force it if the function does return something
+export const functionConfigSchema = Type.Optional(
+  Type.Object({
+    internal: Type.Optional(
+      Type.Object({
+        tags: Type.Object({
+          dev: Type.Boolean(),
+          notice: Type.Boolean(),
+          return: Type.Boolean(),
+        }),
+        inheritdoc: Type.Optional(Type.Boolean()),
+      })
+    ),
+    external: Type.Optional(
+      Type.Object({
+        tags: Type.Object({
+          dev: Type.Boolean(),
+          notice: Type.Boolean(),
+          return: Type.Boolean(),
+        }),
+        inheritdoc: Type.Optional(Type.Boolean()),
+      })
+    ),
+    public: Type.Optional(
+      Type.Object({
+        tags: Type.Object({
+          dev: Type.Boolean(),
+          notice: Type.Boolean(),
+          return: Type.Boolean(),
+        }),
+        inheritdoc: Type.Optional(Type.Boolean()),
+      })
+    ),
+    private: Type.Optional(
+      Type.Object({
+        tags: Type.Object({
+          dev: Type.Boolean(),
+          notice: Type.Boolean(),
+          return: Type.Boolean(),
+        }),
+        inheritdoc: Type.Optional(Type.Boolean()),
+      })
+    ),
+  })
+);
+
+export const configSchema = Type.Object({
+  include: Type.String(),
+  exclude: Type.String({ default: '' }),
+  root: Type.String({ default: './' }),
+  functions: functionConfigSchema,
```

Comment:
I'd also put `constructor: true/false` under `functions` and deprecate `constructorNatspec`.